src/sig_stateful/sig_stfl.h

```diff
@@ -0,0 +1,224 @@
+#define OQS_SIG_STFL_alg_xmss_shake512_h16 "XMSS-SHAKE_16_512"
+/* Algorithm identifier for XMSS-SHAKE_20_512 */
+#define OQS_SIG_STFL_alg_xmss_shake512_h20 "XMSS-SHAKE_20_512"
+/* Algorithm identifier for XMSSMT-SHA2_20/2_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_2_h20 "XMSSMT-SHA2_20/2_256"
+/* Algorithm identifier for XMSSMT-SHA2_20/4_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_4_h20 "XMSSMT-SHA2_20/4_256"
+/* Algorithm identifier for XMSSMT-SHA2_40/2_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_2_h40 "XMSSMT-SHA2_40/2_256"
+/* Algorithm identifier for XMSSMT-SHA2_40/4_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_4_h40 "XMSSMT-SHA2_40/4_256"
+/* Algorithm identifier for XMSSMT-SHA2_40/8_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_8_h40 "XMSSMT-SHA2_40/8_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/3_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_3_h60 "XMSSMT-SHA2_60/3_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/6_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_6_h60 "XMSSMT-SHA2_60/6_256"
+/* Algorithm identifier for XMSSMT-SHA2_60/12_256 */
+#define OQS_SIG_STFL_alg_xmssmt_sha256_12_h60 "XMSSMT-SHA2_60/12_256"
+
+/**
+ * LMS will be added later. So far its only XMSS/XMSS^MT
+ */
+
+
+#define OQS_SIG_STFL_algs_length 20
+
+/**
+ * Returns identifiers for available signature schemes in liboqs.  Used with OQS_SIG_STFL_new.
+ *
+ * Note that algorithm identifiers are present in this list even when the algorithm is disabled
+ * at compile time.
+ *
+ * @param[in] i Index of the algorithm identifier to return, 0 <= i < OQS_SIG_algs_length
+ * @return Algorithm identifier as a string, or NULL.
+ */
+OQS_API const char *OQS_SIG_STFL_alg_identifier(size_t i);
+
+/**
+ * Returns the number of signature mechanisms in liboqs.  They can be enumerated with
+ * OQS_SIG_STFL_alg_identifier.
+ *
+ * Note that some mechanisms may be disabled at compile time.
+ *
+ * @return The number of signature mechanisms.
+ */
+OQS_API int OQS_SIG_STFL_alg_count(void);
+
+/**
+ * Indicates whether the specified algorithm was enabled at compile-time or not.
+ *
+ * @param[in] method_name Name of the desired algorithm; one of the names in `OQS_SIG_STFL_algs`.
+ * @return 1 if enabled, 0 if disabled or not found
+ */
+OQS_API int OQS_SIG_STFL_alg_is_enabled(const char *method_name);
+
+
+/**
+ * @brief SECRET KEY object for stateful signature schemes *
+ */
+typedef struct OQS_SECRET_KEY OQS_SECRET_KEY;
+
+typedef struct OQS_SECRET_KEY {
+
+	/** The (maximum) length, in bytes, of secret keys for this signature scheme. */
+	unsigned long long length_secret_key;
+
+	/** The physical secret key stored in memory as an array of bytes*/
+	uint8_t *secret_key;
+
+	unsigned long long (*sigs_total)(const OQS_SECRET_KEY *secret_key);
+
+	unsigned long long (*sigs_left)(const OQS_SECRET_KEY *secret_key);
+
+	OQS_STATUS (*lock_key)(OQS_SECRET_KEY *sk);
+	OQS_STATUS (*oqs_save_updated_sk_key)(const OQS_SECRET_KEY *sk);
```

Comment:
I don't think it's necessary for this method to start with `oqs_`. None of the others do.